app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -170,9 +177,32 @@ def process_collection(collection, key)
         end
       end
 
-      def gather_data_for_this_region(arm_service, method = "list")
-        @data[:resource_groups].collect do |resource_group|
-          arm_service.send(method, resource_group[:name])
+      # Gather all data for all resource groups. If a subscription ID is provided,
+      # then only collect data for that subscription. Otherwise, collect all data
+      # for all subscriptions.
+      #
+      def gather_data_for_this_region(arm_service, method = 'list_all', subscription_id = nil)
+        subscriptions = subscription_id ? [subscription_id] : @subscriptions
+
+        subscriptions.collect do |sub_id|
+          arm_service.configuration.subscription_id = sub_id
+
+          begin
+            if method.to_s == 'list_all'
+              arm_service.send(method).select { |obj| obj.location == @ems.provider_region }
```

Comment:
In terms of real user time, I typically get better performance with list_all, even when dealing with small numbers of groups for the list method. Here's a quick benchmark and the results I'm seeing against our dev account:
```
groups = %w[miq-azure-test1 miq-azure-test2 miq-azure-test3]

Benchmark.bm(10) do |x|
  x.report("list_all") do
    5.times{ vms.list_all }
  end

  x.report("list") do
    5.times do
      groups.each{ |g| vms.list(g) }
    end
  end
end

                 user     system      total        real
list_all     0.860000   0.020000   0.880000 (  5.463621)
list         0.430000   0.020000   0.450000 (  7.444250)
```